File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/lab/variable.cpp
// variable.cpp

#include <string>
class expression;
#include "expression_visitor.h"
#include "variable.h"

variable::variable(const std::string & a_name)
{
	set_name(a_name);
}

void variable::accept(expression_visitor & a_expression_visitor)
{
	a_expression_visitor.visit(*this);
}

void variable::set_name(const std::string & a_name)
{
	my_name = a_name;
}

const std::string & variable::get_name() const
{
	return my_name;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/lab/Variable.java
// Variable.java

public class Variable extends Terminal
{
	public Variable(String aName)
	{
		setName(aName);
	}

	public void accept(ExpressionVisitor aExpressionVisitor)
	{
		aExpressionVisitor.visit(this);
	}

	private void setName(String aName)
	{
		myName = aName;
	}

	public String getName()
	{
		return myName;
	}

	private String myName;
}
